openhands-sdk/openhands/sdk/agent/agent.py

```diff
@@ -440,4 +447,7 @@ def _execute_action_event(
         # Set conversation state
         if tool.name == FinishTool.name:
             state.agent_status = AgentExecutionStatus.FINISHED
+        elif tool.name == WaitWhileDelegationTool.name:
+            # WaitWhileDelegationTool also sets status to FINISHED to pause main agent
+            state.agent_status = AgentExecutionStatus.FINISHED
```

Comment:
This definitely shouldn't be set to FINISHED, since it's not actually FINISHED. Is PAUSED a status?